sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubClient.cs

```diff
@@ -0,0 +1,282 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Core;
+using Azure.Messaging.EventHubs.Metadata;
+
+namespace Azure.Messaging.EventHubs
+{
+    /// <summary>
+    ///   The main point of interaction with Azure Event Hubs, the client offers a
+    ///   connection to a specific Event Hub within the Event Hubs namespace and offers operations
+    ///   for sending event data, receiving events, and inspecting the connected Event Hub.
+    /// </summary>
+    ///
+    /// <seealso href="https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about" />
+    ///
+    public class EventHubClient
+    {
+        /// <summary>
+        ///   The path of the specific Event Hub that the client is connected to, relative
+        ///   to the Event Hubs namespace that contains it.
+        /// </summary>
+        ///
+        public string EventHubPath { get; protected set; }
+
+        /// <summary>
+        ///   The credential to use for authorization with the Azure Event Hubs connection.  This credential
+        ///   will be validated against the Event Hubs namespace and the requested Event Hub for which operations
+        ///   will be performed.
+        /// </summary>
+        ///
+        public TokenCredential Credential { get; protected set; }
+
+        /// <summary>
+        ///   The set of client options used for creation of this client.
+        /// </summary>
+        ///
+        protected EventHubClientOptions ClientOptions { get; set; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventHubClient"/> class.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string to use for connecting to the Event Hubs namespace; it is expected that the Event Hub path and SAS token are contained in this connection string.</param>
+        ///
+        /// <remarks>
+        ///   If the connection string is copied from the Event Hubs namespace, it will likely not contain the path to the desired Event Hub,
+        ///   which is needed.  In this case, the path can be added manually by adding ";EntityPath=[[ EVENT HUB NAME ]]" to the end of the
+        ///   connection string.  For example, ";EntityPath=telemetry-hub".
+        ///
+        ///   If you have defined a shared access policy directly on the Event Hub itself, then copying the connection string from that
+        ///   Event Hub will result in a connection string that contains the path.
+        /// </remarks>
+        ///
+        public EventHubClient(string connectionString)
+        {
+            EventHubPath = "THIS WOULD BE PARSED FROM THE CONNECTION STRING";
```

Comment:
will this be changed?

Reply:
The parsing, yes.   We'll actually break apart the string into its components;   The constructor itself will be there, it's part of the guidelines covered under "does the service offer a copy/paste connection string from the portal."    I may be misunderstanding the question, though.